### PR TITLE
Specify that Rancher provider is for 1.x only

### PR DESCRIPTION
**Pull request by bradjones1**
### What does this PR do?

Updates the Rancher documentation to specify it is specific to Rancher 1.x.

### Motivation

Rancher 2.x is basically a super-duper reverse proxy and control plane for Kubernetes. 

### More

- [ ] Added/updated tests
- [x] Added/updated documentation

### CODE REVIEW

**Comment by dduportal** [comment]
Hi @bradjones1 , thanks for this contribution.

Could you fix the issues raised by the CI (Travis) related to your change please?

```text
=== Checking HTML content...
- /app/site/providers/rancher/index.html
  *  internally linking to ../kubernetes-crd/index.md, which does not exist (line 1577)
     <a href="../kubernetes-crd/index.md">Kubernetes provider</a>
```

(you can reproduce it locally before pushing your change by following instructions from the page [Contributing -> Documentation](https://docs.traefik.io/v2.0/contributing/documentation/#check-the-documentation), with the target `make -C ./docs/ docs-verify`.

File: docs/content/providers/rancher.md [hunk 1]
```diff
@@ -1,12 +1,14 @@
-# Traefik & Rancher
+# Traefik & Rancher 1.x
```
Comment (dduportal) [inline docs/content/providers/rancher.md]: Could you let the title as it was? As Rancher 2.x is mentioned below thanks to your change, the reader should not be "filtered" at the title, but at the note instead.

File: docs/content/providers/rancher.md [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 
-A Story of Labels, Services & Container
+A Story of Labels, Services & Containers
 {: .subtitle }
 
 ![Rancher](../assets/img/providers/rancher.png)
 
 Attach labels to your services and let Traefik do the rest!
 
+Note: This provider is specific to Rancher 1.x. Rancher 2.x requires Kubernetes and does not have a metadata endpoint of its own for Traefik to query. As such, Rancher 2.x users should utilize the [Kubernetes provider](../kubernetes-crd/index.md) directly.
```
Comment (dduportal) [inline docs/content/providers/rancher.md]:
```suggestion
Note: This provider is specific to Rancher 1.x. Rancher 2.x requires Kubernetes and does not have a metadata endpoint of its own for Traefik to query. As such, Rancher 2.x users should utilize the [Kubernetes provider](./kubernetes-crd.md) directly.
```

Comment (dduportal) [reply docs/content/providers/rancher.md]: Reason: the links shall follow the file structure

```shell
$ pwd
${GOPATH}/src/github.com/containous/traefik/docs/content/providers
$ ls -l ./kubernetes-crd.md 
-rw-r--r--  1 user  user  6856 May 31 10:55 ./kubernetes-crd.md
```

File: docs/content/providers/rancher.md (hunk 1 above)
Comment (bradjones1) [reply docs/content/providers/rancher.md]: I don't feel strongly either way but I think actually filtering out at the heading level makes some sense; If you are on Rancher 2.x you won't need to read through the detail below to know it's not the right place to be.

File: docs/content/providers/rancher.md [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 
-A Story of Labels, Services & Container
+A Story of Labels, Services & Containers
 {: .subtitle }
 
 ![Rancher](../assets/img/providers/rancher.png)
 
 Attach labels to your services and let Traefik do the rest!
 
+Note: This provider is specific to Rancher 1.x. Rancher 2.x requires Kubernetes and does not have a metadata endpoint of its own for Traefik to query. As such, Rancher 2.x users should utilize the [Kubernetes provider](./kubernetes-crd.md) directly.
```
Comment (ldez) [inline docs/content/providers/rancher.md]:
```suggestion
!!! important
    This provider is specific to Rancher 1.x. Rancher 2.x requires Kubernetes and does not have a metadata endpoint of its own for Traefik to query. As such, Rancher 2.x users should utilize the [Kubernetes provider](./kubernetes-crd.md) directly.
```

File: docs/content/providers/rancher.md (hunk 1 above)
Comment (ldez) [reply docs/content/providers/rancher.md]: I agree with @dduportal

Comment (bradjones1) [reply docs/content/providers/rancher.md]: 2 to 1 - resolved. :-)

**Pull request merged**